Sources/MapboxCoreNavigation/Navigator.swift

```diff
@@ -1,7 +1,110 @@
 import Foundation
 import MapboxNavigationNative
+import MapboxDirections
 
 extension Navigator {
+    
+    /**
+     Tiles version string. If not specified explicitly - will be automatically resolved
+     to the latest version.
+     
+     This property can only be modified before creating `Navigator` shared instance, all
+     further changes to this property will have no effect.
+     */
+    static var tilesVersion: String = ""
+    
+    /**
+     A local path to the tiles storage location. If not specified - will be automatically defaulted
+     to the cache subdirectory.
+     
+     This property can only be modified before creating `Navigator` shared instance, all
+     further changes to this property will have no effect.
+     */
+    static var tilesURL: URL? = nil
+    
+    /**
+     Shared instance on `Navigator`.
+     */
+    static let shared: Navigator = {
+        return navigatorWithHistoryRecorder.0
+    }()
+    
+    static func enableHistoryRecorder() throws {
+        try historyRecorder.enable(forEnabled: true)
+    }
+    
+    static func disableHistoryRecorder() throws {
+        try historyRecorder.enable(forEnabled: false)
+    }
+    
+    static func history() throws -> Data {
+        return try historyRecorder.getHistory()
+    }
+    
+    private static var historyRecorder: HistoryRecorderHandle = {
+        return navigatorWithHistoryRecorder.1
+    }()
+    
+    private static let internalQueue = DispatchQueue(label: "com.mapbox.coreNavigation.navigator.internalQueue")
+    
+    /**
+     Provides a new or an existing one `Navigator` instance along with related `HistoryRecorderHandle`,
+     satisfying provided configuration (`tilesVersion` and `tilesURL`).
+     */
+    private static let navigatorWithHistoryRecorder: (Navigator, HistoryRecorderHandle) = {
```

Comment:
I hope that tuple usage is fine in this case. Not sure it's worth creating another entity for this similar to `NavigatorWithHistory`.

Reply:
5ac8c9a5ddebd4f90f49ab391c54ac9ff57707df for #2808 defines a similar struct, NavigatorResources, which is more flexible than a tuple. I think that flexibility will be necessary for #2834.

Reply:
Tuple is no longer relevant after adding `CoreNavigationNavigator.swift` (btw feel free to propose better naming if you have any ideas). Shall I remove `NavigatorResources` in scope of this PR? As I can see it's not used anywhere at the moment.